pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -107,6 +118,27 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, err
 	}
 
+	// Check if the instance was marked to be deleted, and if it is, perform cleanup
+	deleted := instance.GetDeletionTimestamp() != nil
```

Comment:
This block looks similar to what I find in the [user-guide](https://github.com/operator-framework/operator-sdk/blob/master/doc/user-guide.md#handle-cleanup-on-deletion) but for some reason this is hard for me to read. Can we simplify?

```
func Reconcile(...) (...) {
        ...
        if instance.ObjectMeta.DeletionTimestamp.IsZero() {
               // add the finalizer if it's not there                                                          
                if !contains(instance.ObjectMeta.Finalizers, FinalizerName) {                                  
                        instance.ObjectMeta.Finalizers = append(instance.ObjectMeta.Finalizers, FinalizerName)       
                        return reconcile.Result{}, r.client.Update(context.TODO(), instance)                         
                } 
        } else {
                if contains(instance.ObjectMeta.Finalizers, FinalizerName) {                                   
                        // do the cleanup
                        // which can be it's own function if you wish, but for now only really
                        // removes the network-addons-config if certain conditions are met
                        ...
         
                        // remove the finalizer                            
                        instance.ObjectMeta.Finalizers = drop(instance.ObjectMeta.Finalizers, FinalizerName) 
                        return reconcile.Result{}, r.client.Update(context.TODO(), instance) 
                }                                         
        }
        ....
}
```

Reply:
I still think that we should only be worrying about an `hyperconvergeds.hco.kubevirt.io` finalizer...not sure what requires the `foregroundDeletionFinalizer`.

Reply:
I believe this would be easier to follow/maintain.

Reply:
If we don't remove foregroundDeletion, nobody will, it stucks. I can't say I understand where it comes from though.

Reply:
Was trying to be granular with those, for case where we successfully clean-up only half of our CRs. However, that is duplicate of does-CR-still-exist check. I will change it to use a single finalizer.